File: metrics/descriptors/src/main/kotlin/org/convos/metrics/descriptors/navigation/CollectorDelegate.kt
package org.convos.metrics.descriptors.navigation

abstract class CollectorDelegate {
    open fun navigatedTo(source: String, target: String) {}
    open fun presented(source: String, target: String) {}
    open fun closed(screen: String, context: ScreenContext) {}

    open fun identify(userId: String) {}
    open fun updateUserProperties(properties: Map<String, Any?>) {}
    open fun sendEvent(name: String, properties: Map<String, Any?>) {}
}

File: ConvosMetrics/Sources/ConvosMetrics/CollectorDelegate.swift
public struct ScreenContext {
    public let durationSecs: Float

    public init(durationSecs: Float) {
        self.durationSecs = durationSecs
    }
}

open class CollectorDelegate {
    public init() {}

    open func navigatedTo(source: String, target: String) {}
    open func presented(source: String, target: String) {}
    open func closed(screen: String, context: ScreenContext) {}

    open func identify(userId: String) {}
    open func updateUserProperties(properties: [String: Any?]) {}
    open func sendEvent(name: String, properties: [String: Any?]) {}
}
